### PR TITLE
Refactor/customdata docs update

### PR DESCRIPTION
**Pull request by petrKavulok**
This PR
- Prevents passing wrong uri as props to the CustomData component by removing the first character if the first character happens to be a slash '/'
- Documentation was added describing all props necessary and their types

Screenshot 2023-01-05 at 9 46 22

### CODE REVIEW

**Review by aringocode — APPROVED** [review]

File: src/modules/auth/components/CustomDataContainer.js [hunk 1]
```diff
@@ -73,8 +85,12 @@ export function CustomDataContainer({app, resources, customData, setCustomData,
 	const onSave = async (data) => {
 		// first, transform data to appropirate format for api request and skip pairs with empty key field
 		let objToSubmit = await prepForRequest(data);
+		// second, check if uri is in the right format to prevend invalid address
+		let address;
+		uri.charAt(0) === '/' ? address = uri.substring(1, uri.length)
```
Comment (Pe5h4) [inline src/modules/auth/components/CustomDataContainer.js]: @petrKavulok You dont need to have this construction of conditions here at all, you can use regex expression and do simply:

```
let address = url.replace(/^\//g, "");
```

This will always remove the first trailing slash if it will appear in the `url` string. If there will be no trailing slash, the `url` string will stay intact.

You can also make it even simplier by using this in the request (then you dont need to declare `address` variable)

```
let response = await SeaCatAuthAPI.put(`/${url.replace(/^\//g, "")}`, {"data": objToSubmit});
```

File: src/modules/auth/components/CustomDataContainer.js [hunk 2]
```diff
@@ -11,6 +11,18 @@ import {
 	FormGroup
 } from 'reactstrap';
 
+/*
```
Comment (Pe5h4) [inline src/modules/auth/components/CustomDataContainer.js]: @petrKavulok Shouldnt this be in some `doc` file?

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

File: src/modules/auth/components/CustomDataContainer.js (hunk 2 above)
Comment (petrKavulok) [reply src/modules/auth/components/CustomDataContainer.js]: wasn't sure where to put it, since we have docs in `asab-webui/doc` but this is a part of `seacat-admin-webui`. I wasn't sure it was worth creating a `/doc` folder for a documentation of this magnitude. If it should be in the `doc` folder, should the folder be a part of the `/components` folder? or have a different spot in the treemenu?

File: src/modules/auth/tenant/TenantCustomDataContainer.js [hunk 3]
```diff
@@ -1,224 +0,0 @@
-import React, { useState, useEffect } from 'react';
```
Comment (Pe5h4) [inline src/modules/auth/tenant/TenantCustomDataContainer.js:1]: @petrKavulok Why did you remove this container?

File: src/modules/auth/components/CustomDataContainer.js (hunk 2 above)
Comment (Pe5h4) [reply src/modules/auth/components/CustomDataContainer.js]: @petrKavulok Nope, it should be in `doc` folder on top level (there already is a `doc` folder)

```
- asab-webui
- conf
- doc <<<< here is the doc folder where you want to add any broader description regarding of any component
- public
- src
- ....
```

btw. docs for asab-webui surely are in `asab-webui/doc`, but this is not the case, innit?

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

File: src/modules/auth/tenant/TenantCustomDataContainer.js (hunk 3 above)
Comment (petrKavulok) [reply src/modules/auth/tenant/TenantCustomDataContainer.js:1]: `TenantCustomDataContainer.js` was the fist iteration of any custom data container and has been replaced by `CustomDataContainer.js` which is universal.

Comment (Pe5h4) [reply src/modules/auth/tenant/TenantCustomDataContainer.js:1]: I see

File: src/modules/auth/components/CustomDataContainer.js (hunk 2 above)
Comment (petrKavulok) [reply src/modules/auth/components/CustomDataContainer.js]: you're right, I missed that, I'm sorry :-)

File: doc/CustomDataContainer.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# CustomDataContainer
+
+
+`CustomDataContainer` renders a plug'n'play custom data card.
+
+
+## Setup
+
+Before using this component in your project, `react-i18next`, `reactstrap` and `react-hook-form` must be installed and added into the project's `package.json` file:
```
Comment (Pe5h4) [inline doc/CustomDataContainer.md]: @petrKavulok I think here it is not necessary to mention it, since it can be used only within the seacat-admin-ui or where seacat-admin-ui is used as a submodule, but in all cases, you will need to have those libraries installed by default (it is already in package.json). Other case would be if it is a reusable component of asab-webui, but it is not.

File: doc/CustomDataContainer.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+```
+yarn add react-i18next
+yarn add reactstrap
+yarn add react-hook-form
+```
+
+At the moment works ONLY with `apiPath='seacat_auth'`
```
Comment (Pe5h4) [inline doc/CustomDataContainer.md]: @petrKavulok Again, not necessary to mention that, since it is used only within seacat-admin ui and for that you need to have `seacat_auth` proxy set in general

File: doc/CustomDataContainer.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+        ...
+
+		return (
+            <>
+
+                ...
+
+                    <CustomDataaContainer
+						resources={resources}
```
Comment (Pe5h4) [inline doc/CustomDataContainer.md]: @petrKavulok Watch out for padding here, it is not necessary to have it that huge

File: doc/CustomDataContainer.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+```
+import React from 'react';
+import { CustomDataContainer } from '../components/CustomDataContainer';
+
+const YourComponent = (props) => {
+
+        ...
+
+		return (
```
Comment (Pe5h4) [inline doc/CustomDataContainer.md]: @petrKavulok Please format the example a bit better, that it is readable

**Review by Pe5h4 — CHANGES_REQUESTED** [review]
Small remarks. Oh and put the md file to some dedicated `components` dir within the docs

**Review by Pe5h4 — APPROVED** [review]
